walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,198 @@
+  /**
+   * Returns an array with all of the events that meet the requirements set in 
+   *    the predicate.
+   * Time Complexity: O(n^2 * ln(n))
+   *
+   * @param events the array of events that we are going to filter through
+   * @param pred the predicate that will decide whether or not to keep an event
+   * @retunr an array of events that meet the requirements set in the predicate
+   */
+  public static Event[] includeEventIf(Event[] events, Predicate<Event> pred) {
+    ArrayList<Event> filteredEvents = new ArrayList<Event>();
+    for (Event event : events) {
+      // if the events share an attenddee then add it to the array
+      if (pred.test(event)) {
+        filteredEvents.add(event);
+      }
+    }
+    Event[] filteredEventsArray = new Event[filteredEvents.size()];
+    filteredEventsArray = filteredEvents.toArray(filteredEventsArray);
+    return filteredEventsArray;
+  }
+
+  /**
+   * Returns all possible time periods throughout the day when everybody 
+   *    attending this meeting is available. 
+   * Time Complexity: O(n^2 * ln(n))
+   * 
+   * @param events All events that are occurring  
+   * @param request The meeting that the user wants to create and find time for 
+   * @return an array of TimeRange objects
+   */
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) throws Exception{
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // check if duration of meeting is longer than a day or a negative number
+    int durationMeeting = Math.toIntExact(request.getDuration());
+    // long durationMeeting = request.getDuration();
+    if (durationMeeting > 1440 || durationMeeting < 0) {
+      System.err.println("EDGE: duration meeting out of scope");
+      return new ArrayList<TimeRange>();
+    } 
+
+    if (durationMeeting == 0) {
+      System.err.println("EDGE: duration meeting is 0");
+      TimeRange wholeDay = TimeRange.WHOLE_DAY;
+      return new ArrayList<TimeRange>(Arrays.asList(wholeDay));
+    }
+    
+    // Filter events input
+    Event[] eventsArray = new Event[events.size()];
+    eventsArray = events.toArray(eventsArray);
+
+    // remove all events that do not have the attendees from the meeting request 
+    Predicate<Event> isIntersection = new IsIntersection(request.getAttendees());
+    eventsArray = includeEventIf(eventsArray, isIntersection);
+
+    // sort the events that are remaining
+    MergeSort<Event> merge = new MergeSort<Event>();
+    merge.sort(eventsArray, new SortEventsByTime());
+
+    // Get the times of the event and remove events with the same times
+    //    (but keep the longest duration) and the events that are contained in 
+    //    another event
+    ArrayList<int[]> times = new ArrayList<int[]>();
+    try {
+      times = eventToTime(eventsArray);  
+    } catch (Exception e) {
+      String errorMessage = "Error: " + e.getMessage();
+      System.err.println(errorMessage);
+      throw e;
```

Comment:
This shouldn't be necessary as the exception will be printed to the console as it bubbles up to the top level